### PR TITLE
[sdk/go] chore: update some notes and make some lint

### PR DESCRIPTION
**Pull request by asjdf**
I updated some notes, updated the deprecated functions, and make some lint.

Reason for error string change: error string should not be capitalized or end with a punctuation mark.

According to the p/p [readme](https://github.com/pulumi/pulumi#languages), I think it is time to remove some deprecated function like ioutil.ReadFile. #6703

## Checklist

- [x] I have run tests 
- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change
- [x] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: sdk/go/auto/optdestroy/optdestroy.go
Comment (Frassle) [inline sdk/go/auto/optdestroy/optdestroy.go:90]:
```suggestion
// ShowSecrets show config secrets when they appear.
```

File: sdk/dotnet/cmd/pulumi-language-dotnet/main.go
Comment (Frassle) [inline sdk/dotnet/cmd/pulumi-language-dotnet/main.go:375]: I've been wondering about this style myself for a while, @justinvp as you've been around long enough maybe there's a reason we we're doing capitalized error strings?

File: sdk/go/auto/opthistory/opthistory.go
Comment (Frassle) [inline sdk/go/auto/opthistory/opthistory.go:19]:
```suggestion
// ShowSecrets show config secrets when they appear.
```

File: sdk/go/auto/optrefresh/optrefresh.go
Comment (Frassle) [inline sdk/go/auto/optrefresh/optrefresh.go:90]:
```suggestion
// ShowSecrets show config secrets when they appear in the config.
```

File: sdk/go/auto/optup/optup.go
Comment (Frassle) [inline sdk/go/auto/optup/optup.go:118]:
```suggestion
// ShowSecrets show config secrets when they appear.
```

File: sdk/go/common/resource/asset.go
Comment (Frassle) [inline sdk/go/common/resource/asset.go:1054]:
```suggestion
	fh.SetModTime(time.Date(1990, time.January, 1, 0, 0, 0, 0, time.UTC))
```
`SetModTime` handles more than just setting `Modified`

File: sdk/go/common/resource/plugin/context.go
Comment (Frassle) [inline sdk/go/common/resource/plugin/context.go:70]:
```suggestion
// NewContextWithRoot a variation of NewContext that also sets known project Root. Additionally accepts Plugins
```

File: sdk/go/common/resource/plugin/provider.go
Comment (Frassle) [inline sdk/go/common/resource/plugin/provider.go:230]:
```suggestion
// NewDetailedDiffFromObjectDiff computes the detailed diff of Updated, Added and Deleted keys.
```

File: sdk/go/common/resource/plugin/provider_plugin.go
Comment (Frassle) [inline sdk/go/common/resource/plugin/provider_plugin.go:157]:
```suggestion
// NewProviderFromPath create a new provider by loading the plugin binary located at `path`.
```

File: sdk/go/common/resource/properties_diff.go
Comment (Frassle) [inline sdk/go/common/resource/properties_diff.go:58]:
```suggestion
// AnyChanges returns true if there are no changes (adds, deletes, updates) in the diff. Also returns true if
```

File: sdk/go/common/util/cmdutil/spinner.go
Comment (Frassle) [inline sdk/go/common/util/cmdutil/spinner.go:57]:
```suggestion
	// Tick prints the next frame of the spinner. After Tick() has been called, there should be no writes to Stdout before
```

File: sdk/go/common/util/cmdutil/spinner.go
Comment (Frassle) [inline sdk/go/common/util/cmdutil/spinner.go:61]:
```suggestion
	// Reset is called to release ownership of stdout, so others may write to it.
```

File: sdk/go/common/util/gitutil/git.go
Comment (Frassle) [inline sdk/go/common/util/gitutil/git.go:46]: I don't think we're consistent about this for values? I don't even the standard library bothers with repeating names for constant values?

File: sdk/python/python.go
Comment (Frassle) [inline sdk/python/python.go:36]:
```suggestion
// CommandPath finds the correct path and command for Python. If the `PULUMI_PYTHON_CMD`
```

File: sdk/dotnet/cmd/pulumi-language-dotnet/main.go
Comment (justinvp) [reply sdk/dotnet/cmd/pulumi-language-dotnet/main.go:375]: I don't think there's a reason other than the author not knowing or not thinking about the conventions.

Comment (asjdf) [reply sdk/dotnet/cmd/pulumi-language-dotnet/main.go:375]: I was very tangled when I made capitalization changes. I checked the test of the code and didn't see any impact, so I made changes. But worrying about whether it will affect other programs, I don't mind reverting this part of the changes. Thanks for your review.

File: sdk/go/common/util/gitutil/git.go
Comment (asjdf) [reply sdk/go/common/util/gitutil/git.go:46]: In goland it is recommended to add the variable name or function name at the front of the comment so I made these changes. (I think there will be such prompts in other IDEs as well)

File: sdk/go/common/resource/asset.go
Comment (asjdf) [reply sdk/go/common/resource/asset.go:1054]: I made this change due to [this issue](https://github.com/aonez/Keka/issues/422#issuecomment-508575155).

File: sdk/go/common/util/gitutil/git.go
Comment (asjdf) [reply sdk/go/common/util/gitutil/git.go:46]: In fact, I think this code already explains what it is, and the comments here seem completely unnecessary.

Comment (asjdf) [reply sdk/go/common/util/gitutil/git.go:46]: > I don't think we're consistent about this for values? I don't even the standard library bothers with repeating names for constant values?

Can I just remove these unnecessary comments?

File: sdk/go/common/resource/asset.go
Comment (asjdf) [reply sdk/go/common/resource/asset.go:1054]: image
And I thought the package will automatically generate `ModifiedDate` and `ModifiedTime`.

**Review by Frassle — APPROVED** [review]

**Comment by Frassle** [comment]
bors merge

File: sdk/go/auto/local_workspace.go
Comment (AaronFriel) [inline sdk/go/auto/local_workspace.go:465]:
```suggestion
// StackOutputs gets the current set of Stack outputs from the last Stack.Up().
```

File: sdk/dotnet/cmd/pulumi-language-dotnet/main.go
Comment (AaronFriel) [inline sdk/dotnet/cmd/pulumi-language-dotnet/main.go:548]:
```suggestion
// Run is the RPC endpoint for LanguageRuntimeServer::Run
```

File: sdk/go/auto/optdestroy/optdestroy.go
Comment (AaronFriel) [inline sdk/go/auto/optdestroy/optdestroy.go:90]:
```suggestion
// ShowSecrets configures whether to show config secrets when they appear.
```

File: sdk/go/auto/opthistory/opthistory.go
Comment (AaronFriel) [inline sdk/go/auto/opthistory/opthistory.go:19]:
```suggestion
// ShowSecrets configures whether to show config secrets when they appear.
```

File: sdk/go/auto/optrefresh/optrefresh.go
Comment (AaronFriel) [inline sdk/go/auto/optrefresh/optrefresh.go:90]:
```suggestion
// ShowSecrets configures whether to show config secrets when they appear in the config.
```

File: sdk/go/auto/optup/optup.go
Comment (AaronFriel) [inline sdk/go/auto/optup/optup.go:118]:
```suggestion
// ShowSecrets configures whether to show config secrets when they appear.
```

File: sdk/go/auto/workspace.go
Comment (AaronFriel) [inline sdk/go/auto/workspace.go:110]:
```suggestion
	// StackOutputs gets the current set of Stack outputs from the last Stack.Up().
```

File: sdk/go/common/resource/plugin/analyzer_plugin.go
Comment (AaronFriel) [inline sdk/go/common/resource/plugin/analyzer_plugin.go:225]: This seems odd to me, there are cases where pdeps will be empty. This would make this line always allocate a slice?

File: sdk/go/common/resource/plugin/analyzer_plugin.go
Comment (AaronFriel) [inline sdk/go/common/resource/plugin/analyzer_plugin.go:604]: Seems like we should just change this to one line:

```go
		props[k] = s.(JSONSchema)
```

File: sdk/go/common/resource/plugin/context.go
Comment (AaronFriel) [inline sdk/go/common/resource/plugin/context.go:70]:
```suggestion
// NewContextWithRoot is a variation of NewContext that also sets known project Root. Additionally accepts Plugins
```

File: sdk/go/common/resource/plugin/provider_plugin.go
Comment (AaronFriel) [inline sdk/go/common/resource/plugin/provider_plugin.go:157]:
```suggestion
// NewProviderFromPath creates a new provider by loading the plugin binary located at `path`.
```

File: sdk/go/common/util/cmdutil/console.go
Comment (AaronFriel) [inline sdk/go/common/util/cmdutil/console.go:155]: I'm curious why the pointer receiver was dropped here, versus added elsewhere?

File: sdk/go/common/util/cmdutil/trace.go
Comment (AaronFriel) [inline sdk/go/common/util/cmdutil/trace.go:39]: Looks like Go style is to put "Deprecated:" at the beginning of a paragraph in the doc comment:

```go

// TracingToFile indicates if pulumi was called with a file:// scheme URL (--tracing=file:///...). 
//
// Deprecated: Even in this case TracingEndpoint will now have the tcp:// scheme and will point to a
// proxy server that will append traces to the user-specified file. Plugins should respect
// TracingEndpoint and ignore TracingToFile.
```

File: sdk/go/common/util/rpcutil/interceptor.go
Comment (AaronFriel) [inline sdk/go/common/util/rpcutil/interceptor.go:39]:
```suggestion
// OpenTracingStreamServerInterceptor is OpenTracingServerInterceptor for instrumenting streaming gRPC calls.
```

File: sdk/go/common/util/rpcutil/interceptor.go
Comment (AaronFriel) [inline sdk/go/common/util/rpcutil/interceptor.go:64]:
```suggestion
// OpenTracingStreamClientInterceptor is OpenTracingClientInterceptor for streaming gRPC calls.
```

File: sdk/go/pulumi/log.go
Comment (AaronFriel) [inline sdk/go/pulumi/log.go:63]:
```suggestion
// Info logs an informational message that is generally printed to stdout during resource
```

File: sdk/go/pulumi/log.go
Comment (AaronFriel) [inline sdk/go/pulumi/log.go:68]:
```suggestion
// Warn logs a warning to indicate that something went wrong, but not catastrophically so.
```

File: sdk/go/pulumi/mocks.go
Comment (AaronFriel) [inline sdk/go/pulumi/mocks.go:35]:
```suggestion
// MockCallArgs is used to construct a call Mock
```

File: sdk/go/pulumi/types_builtins.go
Comment (AaronFriel) [inline sdk/go/pulumi/types_builtins.go:508]: Was this change run through generate.go? I suspect we'll need to revert here unless you've special cased in generate.go

File: sdk/nodejs/cmd/pulumi-language-nodejs/main.go
Comment (AaronFriel) [inline sdk/nodejs/cmd/pulumi-language-nodejs/main.go:437]:
```suggestion
// Run is the RPC endpoint for LanguageRuntimeServer::Run
```

File: sdk/python/cmd/pulumi-language-python/main.go
Comment (AaronFriel) [inline sdk/python/cmd/pulumi-language-python/main.go:354]: Is this because `plugin` is unexported, making the tag redundant?

**Review by AaronFriel — COMMENTED** [review]
Made a number of suggestions and requested changes.

The change to `types_builtin.go` seems likely to be undone next time the file is generated.

**Comment by AaronFriel** [comment]
@asjdf Thanks for the contribution, sorry for all the late breaking suggestions here, but it felt appropriate to put them in one place.

@Frassle if any of these suggestions seem off, let me know. I think the change to `type_builtins.go` will have to be reverted as I note above, due to it being a generated file? 

Tomorrow I plan to add a `/squash-and-merge` slash command for us to use in cases like this where we might want to apply suggestions. Can we hold off on merging until that's in?

**Review by AaronFriel — CHANGES_REQUESTED** [review]
Requested a change on the types_builtin.go in particular. The remainder of the suggestions I made could be applied in a follow up PR, but this is a convenient place to make them.

**Comment by susanev** [comment]
@asjdf thank you for taking the time to open this pr!! ❤️

File: sdk/python/cmd/pulumi-language-python/main.go
Comment (asjdf) [reply sdk/python/cmd/pulumi-language-python/main.go:354]: yes

File: sdk/go/common/resource/plugin/analyzer_plugin.go
Comment (asjdf) [reply sdk/go/common/resource/plugin/analyzer_plugin.go:225]: In line 221 https://github.com/pulumi/pulumi/blob/dab81be2624df4f4de031a4884deadfd4ef1db1e/sdk/go/common/resource/plugin/analyzer_plugin.go#L221

The program checked lenth of `pd`. The length of `pd` must be greater than 1.

File: sdk/go/common/util/cmdutil/console.go
Comment (asjdf) [reply sdk/go/common/util/cmdutil/console.go:155]: Lint requires the receivers to be of the same type, either all pointer types or all value types.

File: sdk/go/common/util/cmdutil/trace.go
Comment (asjdf) [reply sdk/go/common/util/cmdutil/trace.go:39]: Sorry for my mistake, I'll fix this later. Thanks for your correction.

**Comment by asjdf** [comment]
Thank you all for taking the time to review my pr. ❤

File: sdk/go/common/resource/plugin/analyzer_plugin.go
Comment (asjdf) [reply sdk/go/common/resource/plugin/analyzer_plugin.go:604]: Thanks for the advice, it looks like my mind is out.

**Comment by asjdf** [comment]
> Requested a change on the types_builtin.go in particular. The remainder of the suggestions I made could be applied in a follow up PR, but this is a convenient place to make them.

I think you are right, so I revert the change in types_builtin.go. I'm sorry for my hasty changes in types_builtin.go.

**Review by AaronFriel — APPROVED** [review]
Looks good to me, thank you!

bors merge

**Comment by AaronFriel** [comment]
@asjdf I'll need to take another pass, as it looks tests are failing with an error that points to a string formatting change:


> error: failed to load language plugin : Could not automatically download and install language plugin 'pulumi-language-', install the plugin using `pulumi plugin install language `.
Underlying error: 403 HTTP error fetching plugin from https://api.github.com/repos/pulumi/pulumi-/releases/latest

**Comment by asjdf** [comment]
> @asjdf I'll need to take another pass, as it looks tests are failing with an error that points to a string formatting change:
> 
> 
> 
> 
> 
> > error: failed to load language plugin : Could not automatically download and install language plugin 'pulumi-language-', install the plugin using `pulumi plugin install language `.
> 
> Underlying error: 403 HTTP error fetching plugin from https://api.github.com/repos/pulumi/pulumi-/releases/latest

I'm really sorry that this pr has taken up so much of your time. Thank you very much for your help.❤️

File: sdk/go/common/util/cmdutil/console.go
Comment (AaronFriel) [reply sdk/go/common/util/cmdutil/console.go:155]: Which linter are you using? I think it's different from our style guide and that rule causes issues with marshaling. See:

https://go.dev/play/p/w7fGZQBjmp6

For example, in `sdk/go/common/resource/config/key.go`, the tests fail unless `MarshalJJSON` is a value receiver and `UnmarshalJSON` is a pointer receiver.

**Comment by AaronFriel** [comment]
@asjdf thank you for taking the time to make this contribution! It's long overdue cleanup to be consistent with Go's style guide.

I've started a new PR #11002 based off this branch to see why those some of those checks failed. It looks like the main issue is Go's behavior and requirements for pointer vs value receivers with marshaling.

I found [this Go playground entry](https://go.dev/play/p/w7fGZQBjmp6) (the same as linked above) which shows that Go's behavior here seems inconsistent. I learned the hard way to be careful changing methods from pointer receivers to value receivers early on.

**Comment by asjdf** [comment]
> @asjdf感谢您抽出宝贵时间做出贡献！为了与 Go 的风格指南保持一致，早就应该进行清理了。
> 
> 我已经基于此分支启动了一个新的 PR #11002，以了解其中一些检查失败的原因。看起来主要问题是 Go 对带有编组的指针与值接收器的行为和要求。
> 
> 我发现[这个 Go 游乐场条目](https://go.dev/play/p/w7fGZQBjmp6)（与上面链接的相同）表明 Go 在这里的行为似乎不一致。我很早就学会了小心地将方法从指针接收器更改为值接收器的艰难方法。



> @asjdf thank you for taking the time to make this contribution! It's long overdue cleanup to be consistent with Go's style guide.
> 
> I've started a new PR #11002 based off this branch to see why those some of those checks failed. It looks like the main issue is Go's behavior and requirements for pointer vs value receivers with marshaling.
> 
> I found [this Go playground entry](https://go.dev/play/p/w7fGZQBjmp6) (the same as linked above) which shows that Go's behavior here seems inconsistent. I learned the hard way to be careful changing methods from pointer receivers to value receivers early on.

Thank you for your help and looks like I can turn this pr off?

**Comment by AaronFriel** [comment]
@asjdf if you want to click "Allow edits from maintainers", I can push my changes back onto this branch and merge.

**Comment by AaronFriel** [comment]
Closing in favor of #11002

**Pull request closed without merging**
